File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
    
      int flag = 0;

      for(int x = 0; x < prices.length; x++){
        for(int y = x+1; y<prices.length; y++){
            if(prices[y]-prices[x] > flag){
                flag = prices[y]- prices[x];
            }
        }
      }
      return flag;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        memo = {}

        def dp(i, holding):
            if i >= len(prices):
                return 0
            if (i, holding) in memo:
                return memo[(i, holding)]
            
            buy = -prices[i] + dp(i+1, True)
            sell = prices[i] + dp(i+1, False)
            skip = dp(i+1, holding)

            if holding:
                memo[(i, holding)] = max(sell, skip)
                return memo[(i, holding)]
            else:
                memo[(i, holding)] = max(buy, skip)
                return memo[(i, holding)]
        return dp(0, False)

            
            